packages/trader/src/Modules/Account/Sections/Profile/FinancialAssessment/financial-assessment.jsx

```diff
@@ -141,24 +142,32 @@ class FinancialAssessment extends React.Component {
     };
 
     componentDidMount() {
+        this.is_mounted = true;
```

Comment:
The code inside `componentDidMount` run when component is mounted! why we need another flag for checking this inside `componentDidMount` one more time?!

Reply:
@mahdiaryayi-fs There are times where API response comes much later (when the component has already unmounted - switching to a different page after sending an API call for example). Had to add this check so it doesn't run `setState` when it is already unmounted. You can refer to [L150](https://github.com/binary-com/deriv-app/pull/948/files#diff-3768d17628d6a7d7dc0d10b18c894880R150) here. Any idea on how to better handle this ?

Reply:
@kelly-binary What you have done is true and we can keep it this way. one more fancy way for managing these cases is using a [cancelable promise](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html). but what you have done is true too. I approved it

Reply:
@mahdiaryayi-fs actually using a cancelable promise is a much cleaner workaround. we can add it to code improvement project perhaps? 😄

Reply:
If you want to keep it for now, you may change your TODO comment and refer to this cancelable promise for future reference.